### DISCUSSION TITLE
Conversion of polar to Cartesian and back (with filtering in between)

### DISCUSSION THREAD

**Discussion by DavidAckerman88** [ImageMagick/ImageMagick#7280] [Help]
### ImageMagick version

7.1.1.1-31

### Operating system, version and so on

MacOS 14.1.2

### Description

Total neophyte on ImageMagick searching for a high-fidelity cartesian -> polar warp that I will filter (to create radial blur) and then convert back to cartesian to subtract from the original to create radial high-pass images.  (All this for eclipse corona processing).  Works for .jpgs but not for 16-bit .tiff files yet.  Please recommend a tutorial.

**Comment by fmw42** [comment]
Read https://imagemagick.org/Usage/distorts/#polar and the sections that follow about depolarization and the full cycle polar and dipolar. All for command line Imagemagick.  I know nothing about eclipse corona.

**Comment by snibgo** [comment]
The IM operations would be "`-distort polar`" and "`-distort depolar`". See https://imagemagick.org/script/command-line-options.php#distort

For a tutorial, [Polar distortions](http://im.snibgo.com/poldist.htm) may be useful.

**Reply by DavidAckerman88** [reply]
Follow-on question -- is it possible to specify a pair of center coordinates in using polarblur?  Or should I simply pad my image so that the desired center of the polar blur ends up in the center of the padded image?

**Reply by snibgo** [reply]
What is "polarblur"? That isn't an IM operation.

For "`-distort polar`" and "`-distort depolar`", yes, they can have specified centres. See https://imagemagick.org/script/command-line-options.php#distort

**Comment by DavidAckerman88** [comment]
Thank you for your instant response.  That will keep me coming back and
maybe I'll learn imageMagick.  I do like command line programs!

All the best,
David

PS:  I do understand that you might not know about the eclipse corona but I
do and you have helped me continue to refine my raw photos of it.  In the
end, if I get what I hope to, I'll send you a copy in case you want to
showcase just another thing your IM can do.

**Comment by DavidAckerman88** [comment]
Hi again -- and thank you for your quick reply.

I am confused between the scripts listed here:
http://www.fmwconcepts.com/imagemagick/index.php and the commands that
include -distort polar and -distort depolar.  I do understand that the
scripts on Fred's ImageMagick Scripts page use the lower level commands but
how exactly to use them on the command line is something I'm still working
out.  Moreover, calling them from a high level scripting language, e.g.,
Matlab, is also something that I'm working on.

So, polarblur, which does a rect2polar transform followed by a 1 or 2D
Gaussian blur in the polar domain and the transforms back using polar2rect
would be very useful to me if I can wrangle it to do what I want,

**Reply by fmw42** [reply]
You do not need my scripts and they do not do what you are asking. 

```
magick input.suffix \
-distort DePolar -1  \
-blur 0xsigma \
-virtual-pixel HorizontalTile -background black \
-distort  Polar  -1  \
output.suffix

```
Use whatever value you want for sigma above. That is replace the word sigma with your value.  User more depolar/polar distortion arguments as you want: Radius_Max Radius_Min Center_X,Center_Y Start_Angle,End_Angle

See https://imagemagick.org/Usage/distorts/#polar
